### PR TITLE
Add precompile contracts' addresses to the access list

### PR DESCRIPTION
**Pull request by ChihChengLiang**
### Description

We upstream scroll's precompile handling in bus mapping so that the EVM circuit warms the precompile addresses at the beginning of the transaction.

### Issue Link

#1626

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Content

- Warm precompile addresses
- Fix some minor issues in StepRws offset and the Rws debugging messages.

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/util.rs [hunk 1]
```diff
@@ -475,7 +475,7 @@ impl<'a> StepRws<'a> {
     }
     /// Increment the step rw operation offset by `offset`.
```
[the remark this answers is not in the document]
Comment (ChihChengLiang) [reply zkevm-circuits/src/evm_circuit/util.rs]: Addressed in [8126290](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1640/commits/8126290e6051be9d79270b787e2c0c56760d6bee)

File: bus-mapping/src/evm/opcodes/begin_end_tx.rs [hunk 2]
```diff
@@ -60,6 +60,19 @@ fn gen_begin_tx_steps(state: &mut CircuitInputStateRef) -> Result<ExecStep, Erro
         false,
     )?;
 
+    // Add precompile contract address to access list
+    for address in 1..=9 {
```
[the remark this answers is not in the document]
Comment (ChihChengLiang) [reply bus-mapping/src/evm/opcodes/begin_end_tx.rs]: Addressed in [d9f03ab](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1640/commits/d9f03ab0bf4cccdd7a30b149b2de3df8b70f82b9)

File: bus-mapping/src/evm/opcodes/begin_end_tx.rs [hunk 3]
```diff
@@ -11,6 +11,8 @@ use eth_types::{
 };
 use ethers_core::utils::get_contract_address;
 
+const PRECOMPILE_COUNT: u64 = 9;
```
Comment (KimiWu123) [inline bus-mapping/src/evm/opcodes/begin_end_tx.rs:14]: minor suggestion: move `PRECOMPILE_COUNT` declaration to eth-types/src/evm_types.rs then we don't need to define `PRECOMPILE_COUNT` in different files.

**Review by KimiWu123 — APPROVED** [review]
LGTM!

File: bus-mapping/src/evm/opcodes/begin_end_tx.rs (hunk 3 above)
Comment (ChihChengLiang) [reply bus-mapping/src/evm/opcodes/begin_end_tx.rs]: Good point. I addressed it in [bb22082](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1640/commits/bb220825358c4a6203998245d7f7e0e302821b98)

**Pull request merged**